zkevm-circuits/src/rlp_circuit_fsm.rs

```diff
@@ -178,6 +179,53 @@ impl RlpFsmRomTable {
     }
 }
 
+/// Data decoding table simulates a stack-like structure for constraining remaining_bytes
+#[derive(Clone, Copy, Debug)]
+pub struct RlpDecodingTable {
+    /// Is Write to decoding stack
+    pub is_write: Column<Advice>,
+    /// Key1 (Id), concat of tx_id, format
+    pub id: Column<Advice>,
+    /// Key2 (Address), in this case depth
+    pub depth: Column<Advice>,
+    /// Value
+    pub value: Column<Advice>,
+    /// Value Previous
+    pub value_prev: Column<Advice>,
+    /// Stack Accumulator
+    /// accumulates remaining bytes on each depth level (excluding top of stack)
+    pub stack_acc: Column<Advice>,
+    /// Power of rand for stack accumulator on depth level (address)
+    pub stack_acc_pow_of_rand: Column<Advice>,
```

Comment:
Why can't we sort the rows for operating with `remining_bytes` at depth `d` in rlp decoding table?

Reply:
Currently, the assumption on access list is:

(i) There's no prior knowledge on how long the access list might be. The length is dynamic.

For the stack construct:

In the `rw` stack construct, the main objective is to ensure that a value pushed and popped immediately after from the same depth level is the same. For RLP, however, there's another important constraint. The value pushed to a higher depth is related to the updates on the lower depth. For example, supposed at depth 2 we have 1000 bytes left and encounter a nested access list. Then a portion of that 1000 bytes will be pushed to depth 3 (let's say 800 bytes as an example). In this case, there're two stack operations. First update depth 2 to 200 bytes, then push 800 bytes to depth 3. The `PUSH` operation, in other words, has a direct correspondence with the `UPDATE` operation on the lower depth (unlike in `rw` stack case, where values on each depth are independent). This relation necessitates a constraint across different depth level.

Here's the problem:

Due to assumption (i), IF the rows are sorted by depth, it's uncertain how to exact the relative location of these rows (on different depth levels). One might consider that since access list is the only nested structure right now, a fixed rotation might be possible to accomplish the purpose, but if any other nested structures are introduced in the future, this mechanism will readily break down. 

Current Codebase:

Instead, the current design positions the `PUSH` for a higher depth immediately after the `UPDATE` for the lower depth, with `stack_acc` carrying a summary for all bytes left on lower depth levels. This mechanism fixes the positions of cross-depth rows, making the objective of having constraints across depth possible. 

cc @roynalnaruto

Reply:
> The value pushed to a higher depth is related to the updates on the lower depth. The PUSH operation, in other words, has a direct correspondence with the UPDATE operation on the lower depth (unlike in rw stack case, where values on each depth are independent). 

Before I read the codes, I thought the RLP finite state machine would do lookup into the `RlpDecodingTable` and we sort rows of `RlpDecodingTable` by the lexicographical order of `(tx_id, format, al_idx, sk_idx)`.

Then in the above example, we "would" make two lookups to the `RlpDecodingTable` with input exprs as
1. `(tx_id, format, al_idx = 2, sk_idx, depth, value=value_prev - 800, value_prev)`;
2. `(tx_id, format, al_idx = 2+1, sk_idx = 0, depth + 1, value=800, value_prev=0)`;

Reply:
The `stack_acc` mechanism is removed from the newest commits. Lookups are used instead.